src/SSW.SophieBot/components/SSWSophieBot.Components/ComponentHelper.cs

```diff
@@ -7,9 +7,9 @@ public static class ComponentHelper
     {
         public static DateTime ToUserLocalTime(this DateTime dateTime, DialogContext dc)
         {
-            var serverLocalTime = DateTime.SpecifyKind(dateTime, DateTimeKind.Utc);
+            var utcTime = dateTime.ToUniversalTime();
             var utcOffset = dc.Context.Activity.LocalTimestamp.GetValueOrDefault().Offset;
-            return serverLocalTime.Subtract(utcOffset);
+            return utcTime.Add(utcOffset);
```

Comment:
There are actually two cases about date time (Sorry I didn't pay attention before)
1. ToUserLocalTime - e.g. Displaying a UTC time on user end
2. From UserLocalTime - e.g. Receiving a date string from user end and parse to UTC time for filtering

Reply:
See my latest commit

Reply:
Test pass in dev

![image](https://user-images.githubusercontent.com/37203901/142804542-661fbe31-1152-4804-b98d-61e03bebd178.png)
**Figure: Date in Teams matches the current date**